### PR TITLE
App Config Min Feature Flags + Change Log and Readme Updated

### PR DESCRIPTION
**Pull request by mrm9084**
# Description
Updating Feature Flags for future plus updating change log and read me

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/appconfiguration/azure-appconfiguration-provider/README.md [hunk 1]
```diff
@@ -60,8 +60,7 @@ You can refine or expand the configurations loaded from your store by using `Set
 
 ```python
 selects = {SettingSelector(key_filter="*", label_filter="\0"), SettingSelector(key_filter="*", label_filter="dev")}
-config = AzureAppConfigurationProvider.load(
-    endpoint=endpoint, credential=default_credential, selects=selects)
+config = load_provider(endpoint=endpoint, credential=default_credential, selects=selects)
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration-provider/README.md:63]: Where is default_credential defined?

File: sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py [hunk 2]
```diff
@@ -129,7 +130,7 @@ def load_provider(*args, **kwargs) -> "AzureAppConfigurationProvider":
                 provider._dict[trimmed_key] = secret
             elif isinstance(config, FeatureFlagConfigurationSetting):
                 feature_management = provider._dict.get(FEATURE_MANAGEMENT_KEY, {})
-                feature_management[trimmed_key] = config.value
+                feature_management[trimmed_key[len(FEATURE_FLAG_PREFIX) :]] = config.value
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py:133]: len(FEATURE_FLAG_PREFIX) is the length of the string. Wanted to double check if this is the design.

Comment (xiangyan99) [reply sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py:133]: Looks like you want to remove the prefix ".appconfig.featureflag/". 

Do you need to check if it starts with it?

File: sdk/appconfiguration/azure-appconfiguration-provider/README.md (hunk 1 above)
Comment (mrm9084) [reply sdk/appconfiguration/azure-appconfiguration-provider/README.md:63]: No where here, I think the README assumes that they know how to make one. Should I add it?

File: sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py (hunk 2 above)
Comment (mrm9084) [reply sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py:133]: It "should" always have it. But added a check to be safe.

File: sdk/appconfiguration/azure-appconfiguration-provider/README.md (hunk 1 above)
Comment (xiangyan99) [reply sdk/appconfiguration/azure-appconfiguration-provider/README.md:63]: We received some feedback reporting that it was confusing with such stuff. :)

Maybe you can just use DefaultAzureCredential().

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**